### PR TITLE
docs(flink): add to the set of documented backends

### LINKED ISSUE #7737: docs(flink): add to the set of documented backends
### Please describe the issue

As a prerequisite to officially releasing the Flink backend, we need to add a simple documentation page, as we have for all of the other backends.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by deepyaman**
Closes #7737

### CODE REVIEW

File: docs/backends/pyspark.qmd [hunk 1]
```diff
@@ -81,7 +79,7 @@ con = ibis.pyspark.connect(session=session)
 :::
 
 ::: {.callout-note}
-The `pyspark` backend does not create `SparkSession` objects, you must create a `SparkSession` and pass that to `ibis.pyspark.connect`.
```
Comment (deepyaman) [inline docs/backends/pyspark.qmd:84]: Technically it can, though, using `from _url`...

Comment (cpcloud) [reply docs/backends/pyspark.qmd:84]: Yeah, this document is bit outdated.

**Comment by cpcloud** [comment]
`/preview` won't work anymore, it was a bit of a security hole so we now have to add an issue label to run it which requires at least "Maintain" permissions to add.

**Comment by cpcloud** [comment]
It looks like you changed something that requires re-rendering. Are you able to run the docs build locally?

**Comment by deepyaman** [comment]
> It looks like you changed something that requires re-rendering. Are you able to run the docs build locally?

I hadn't installed Quarto yet, but looks like I should get to doing that...

File: docs/backends/pyspark.qmd [hunk 2]
```diff
@@ -81,7 +79,7 @@ con = ibis.pyspark.connect(session=session)
...
+The `pyspark` backend does not create `SparkSession` objects; you must create a `SparkSession` and pass that to `ibis.pyspark.connect`.
```
Comment (deepyaman) [inline docs/backends/pyspark.qmd:82]:
```suggestion
The `pyspark` backend does not create `SparkSession` objects (unless you [connect using a URL](#ibis.connect-url-format)); you must create a `SparkSession` and pass that to `ibis.pyspark.connect`.
```
@cpcloud maybe this is slightly more accurate?

**Comment by deepyaman** [comment]
> /preview

image

**Review by cpcloud — APPROVED** [review]

**Pull request merged**